### PR TITLE
Added CentOS 7 support. Added optional ipset and gd support

### PR DESCRIPTION
**Pull request by bvansomeren**
Hi James,

First of all, many thanks for creating this role. I had something local, but it was not as elegant as yours (I especially like how you handled the configuration for csf.conf!)

I saw you added initial support for RHEL / CentOS 7 and I hope you didn't mind me forking from your project and adding a few minor additions to make things work on CentOS 7.
Have a look at my pull request and if you need me to make any changes, please let me know.

I'm a bit new at this Github thing and I've tried to keep things in the same style as you had.
I also checked within Vagrant that the project still works with Ubuntu.

A short list:
- Remove firewalld on RHEL / CentOS 7. And install deps
- Allow install of ipset (default off)
- Allow install of libgd-perl for statistical screens (default off)
- Minor fix where the csf / lfd restart required sudo

Thanks again!

Regards,

Barry

### CODE REVIEW

**Comment by jloh** [comment]
Hey Barry! Awesome pull request, I'll do some more checking later today and pull.

Thanks!

**Comment by jloh** [comment]
On second thought, we should really add some more tests to the test Matrix for the new options you've added. ~~Do you want to have a go at that or want me to add them?~~

If you checkout the [`more_travis`](https://github.com/Mooash/csf-ansible-role/blob/more_travis/.travis.yml) branch you can see what I intend. What do you think?

**Comment by bvansomeren** [comment]
Hi James,

That looks good, as a Jenkins user I'm starting to like Travis. 
You want me to take your changes into my pull request or will you do the merge?

**Comment by jloh** [comment]
You should make the changes and do a rebase of your PR since you've taken the time to do this work.

If you could make the changes to the `.travis-ci.yml` file, commit and rebase your PR that'd be great.

See [here](http://stackoverflow.com/questions/14534397/squash-all-my-commits-into-one-for-github-pull-request/16642628#16642628) if you don't know how to rebase.

**Comment by bvansomeren** [comment]
Hi James,

I'm afraid my experience has been mostly with SVN and I'm not used to thinking in "4 dimensions" just yet; I hate to ask for an ELI5 (Explain it Like I'm 5) on how to do this, but it's a little unclear how I can rebase my pull request on your new branch called "more_travis"
Would it make sense that either I manually add your changes to my pull request or that I refork and redo the changes?

Thanks!

**Comment by jloh** [comment]
Sorry, I explained that very poorly. If you just look at the changes I've made to the `.travis-ci.yml` file in the "more_travis" branch and manually add those changes to the pull request that'd be great.

If you are then able to do a rebase (don't worry if you can't) that'd be awesome.

**Comment by bvansomeren** [comment]
Hi James,

No worries and thank you for clearing that up; I've gone ahead and amended my pull request; I'm going to go and dig out that Git book for over the weekend.

I might also have more extra's for this CSF role in the future; I'd love to share those as well

Regards,

Barry

**Comment by jloh** [comment]
Thanks @bvansomeren. All merged!

**Comment by bvansomeren** [comment]
Thanks! :)
